foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/queries/InterfaceContainerPolicy.java

```diff
@@ -169,26 +175,77 @@ public Method getCloneMethod() {
         return cloneMethod;
     }
 
+    private static final class ClassWeakReference extends WeakReference<Class> {
```

Comment:
Is there a specific reason we can't just use ```WeakReference<Class>``` instead of creating this new class? Im guessing it's something to do with overriding equals, but the Object.equals implementation should work, right?

Reply:
The equals on WeakReference is Object.equals which is an == check.  That will never happen which means we will never find anything in the ConcurrentHashMap.